src/pystatis/db.py

```diff
@@ -28,6 +29,35 @@ def identify_db(name: str) -> list[str]:
     return db_match
 
 
+def identify_db(table_name: str) -> str:
+    """Identify database by matching with the provided item code.
+    This is done by matching the item code to the database regex. If multiple matches
+    exist, use the first with existing credentials.
+    In this case, it must be a Cube (provided all regexing works as intended).
+
+    Args:
+        table_name (str): Query parameter 'name' corresponding to the item code.
+
+    Returns:
+        db_name (str): Identified database.
+    """
+    db_match = identify_db_matches(table_name)
+
+    if db_match:
+        for db_name in db_match:
+            # Return first hit with existing credentials.
+            if check_credentials(db_name):
+                return db_name
+        raise PystatisConfigError(
```

Comment:
why do we need this check in this function? This function should only return a database given a string, but it actually checks if the credentials are valid...so it does not really identify the database. I thought this check is done in a different place, isn't it? I don't think we need it here?

Reply:
and the `raise` command could be put in a `else` branch which makes it even clearer that this code is only execute when the for loop ends without any match

Reply:
The logic is the following:

1. Find all possible DBs, given the table name.
2. Go through all DB candidates and check if the user provided credentials for a candidate. If so: return that as the matched DB